python/python/tests/test_async_client.py

```diff
@@ -4453,6 +4453,123 @@ async def test_object_refcount(self, redis_client: TRedisClient):
+    async def test_function_load_standalone(self, redis_client: RedisClient):
+        # TODO: Test function with FCALL
+        # TODO: Test with FUNCTION LIST
+        min_version = "7.0.0"
+        if await check_if_server_version_lt(redis_client, min_version):
+            return pytest.mark.skip(reason=f"Redis version required >= {min_version}")
+
+        lib_name = f"mylib1C{get_random_string(5)}"
+        code = f"#!lua name={lib_name}\n redis.register_function('myfunc1c{get_random_string(5)}', function(keys, args) return args[1] end)"
+
+        assert await redis_client.function_load(code) == lib_name
+
+        # re-load library without overwriting
+        with pytest.raises(RequestError) as excinfo:
+            await redis_client.function_load(code)
+
+        assert "Library '" + lib_name + "' already exists" in str(excinfo.value)
+
+        # re-load library with overwriting
+        assert await redis_client.function_load(code, True) == lib_name
+
+        new_code = f"""{code}\n redis.register_function('myfunc2c{get_random_string(5)}', function(keys, args) return #args end)"""
+
+        assert await redis_client.function_load(new_code, True) == lib_name
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    @pytest.mark.parametrize("with_route", [True, False])
+    async def test_function_load_cluster(
+        self, redis_client: RedisClusterClient, with_route: bool
+    ):
+        # TODO: Test function with FCALL
+        # TODO: Test with FUNCTION LIST
+        min_version = "7.0.0"
+        if await check_if_server_version_lt(redis_client, min_version):
+            return pytest.mark.skip(reason=f"Redis version required >= {min_version}")
+
+        lib_name = f"mylib1C{with_route}{get_random_string(5)}"
+        code = f"#!lua name={lib_name}\n redis.register_function('myfunc1c{with_route}{get_random_string(5)}', function(keys, args) return args[1] end)"
+
+        route = SlotKeyRoute(SlotType.PRIMARY, "1") if with_route else None
+
+        assert await redis_client.function_load(code, False, route) == lib_name
+
+        # re-load library without overwriting
+        with pytest.raises(RequestError) as e:
+            await redis_client.function_load(code, False, route)
+        assert "Library '" + lib_name + "' already exists" in str(e)
+
+        # re-load library with overwriting
+        assert await redis_client.function_load(code, True, route) == lib_name
+
+        new_code = f"{code}\n redis.register_function('myfunc2c{get_random_string(5)}', function(keys, args) return #args end)"
+
+        assert await redis_client.function_load(new_code, True, route) == lib_name
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    @pytest.mark.parametrize("with_route", [True, False])
+    async def test_function_flush_cluster(
```

Comment:
Combine tests for flush and load
If no - apply comments to function load tests here too